base_view_inheritance_extension/tests/test_base_view_inheritance_extension.py

```diff
@@ -0,0 +1,243 @@
+            <field name="account_move_id" position="attributes">
+                <attribute name="context" operation="update">
+                    {"default_company_id": company_id}
+                </attribute>
+            </field>
+            """
+        )
+        res = self.env["ir.ui.view"].apply_inheritance_specs(source, specs)
+        self.assertEqual(
+            res.xpath('//field[@name="account_move_id"]')[0].attrib["context"],
+            "{'default_journal_id': journal_id, 'default_company_id': company_id}",
+        )
+
+    def test_update_context_complex(self):
+        source = etree.fromstring(
+            """
+            <form>
+                <field
+                    name="invoice_line_ids"
+                    context="{
+                        'default_type': context.get('default_type'),
+                        'journal_id': journal_id,
+                        'default_partner_id': commercial_partner_id,
+                        'default_currency_id': (
+                            currency_id != company_currency_id and currency_id or False
+                        ),
+                        'default_name': 'The company name',
+                    }"
+                />
+            </form>
+            """
+        )
+        specs = etree.fromstring(
+            """
+            <field name="invoice_line_ids" position="attributes">
+                <attribute name="context" operation="update">
+                    {
+                        "default_product_id": product_id,
+                        "default_cost_center_id": (
+                            context.get("handle_mrp_cost") and cost_center_id or False
+                        ),
+                    }
+                </attribute>
+            </field>
+            """
+        )
+        res = self.env["ir.ui.view"].apply_inheritance_specs(source, specs)
+        expected_items = [
+            "'default_type': context.get('default_type')",
+            "'journal_id': journal_id",
+            "'default_partner_id': commercial_partner_id",
+            (
+                "'default_currency_id': "
+                "currency_id != company_currency_id and currency_id or False"
+            ),
+            "'default_name': 'The company name'",
+            "'default_product_id': product_id",
+            (
+                "'default_cost_center_id': "
+                "context.get('handle_mrp_cost') and cost_center_id or False"
+            ),
+        ]
+        self.assertEqual(
+            res.xpath('//field[@name="invoice_line_ids"]')[0].attrib["context"],
+            "{%s}" % ", ".join(expected_items),  # noqa: UP031
+        )
+
+    def test_update_attrs_new_key(self):
+        """Test that we can add new keys to an existing dict"""
+        source = etree.fromstring(
+            """
+            <form>
+                <field
+                    name="ref"
+                    attrs="{'invisible': [('state', '=', 'draft')]}"
```

Comment:
This test makes no sense? - attrs is gone in v18.  Hmm, this might have big consequences for module - all these tests containing attrs I think will need changing.